### PR TITLE
Pre features

### PR DESCRIPTION
**Pull request by wuyepabdul**
In this milestone, we did the following:
- Created schema_based_on_diagram.sql file
- Implemented the database from the given database schema diagram.
- Add foreign key constraints on medical_histories, invoices, and invoice_items tables

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    unit_price decimal(10, 2),
+    quantity int,
+    total_price decimal,
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount decimal(10, 2),
+    generated_at int,
```
Comment (descholar-ceo) [inline schema_based_on_diagram.sql]: - [x] generated_at is supposed to be a timestamp instead of being an int

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    PRIMARY KEY (id)
+);
+
+ALTER TABLE medical_histories ADD FOREIGN KEY patient_id REFERENCES patients(id);
+
+ALTER TABLE invoices ADD FOREIGN KEY medical_history_id REFERENCES medical_histories(id);
+
+ALTER TABLE invoice_items ADD FOREIGN KEY invoice_id REFERENCES invoice_id(id);
+
+ALTER TABLE invoice_items ADD FOREIGN KEY treatment_id REFERENCES treatments(id);
```
Comment (descholar-ceo) [inline schema_based_on_diagram.sql]: - [x] We are missing one more table: all "join table" for the many-to-many relastionships should be created, even if they are not showed on the diagram. There fore you should create a table between medical_histories and treatments  (name can be different)

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @wuyepabdul and @Mhdez221993 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (descholar-ceo) [reply schema_based_on_diagram.sql]: :+1: Addressed :heavy_check_mark:

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (descholar-ceo) [reply schema_based_on_diagram.sql]: :+1: Addressed :heavy_check_mark:

**Review by descholar-ceo — APPROVED** [review]
Hi @wuyepabdul and @Mhdez221993 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wuyepabdul** [comment]
Thanks for the review @descholar-ceo

**Pull request merged**